bundles/org.openhab.core.voice/src/main/java/org/openhab/core/voice/internal/VoiceManagerImpl.java

```diff
@@ -863,4 +879,12 @@ private Comparator<Voice> createVoiceComparator(Locale locale) {
         }
         return null;
     }
+
+    private void cleanSingleDialogProcessors() {
+        singleDialogProcessors.forEach((key, processor) -> {
+            if (!processor.isProcessing()) {
+                singleDialogProcessors.remove(key);
+            }
+        });
+    }
```

Comment:
I don't think you need a method here, it's a one-liner:

```
    singleDialogProcessors.removeIf(e -> !e.getValue().isProcessing());
```

Reply:
Ends up like this:
```
singleDialogProcessors.values().removeIf(e -> !e.isProcessing());
```
I didn't know about the entrySet and values methods retuning views that you can use to modify the initial collection. Really good to know, thank you!